### PR TITLE
feat(ehr): patch patient routes

### PR DESCRIPTION
**Pull request by thomasyopes**
Ref: #1040

### Dependencies

- Upstream: https://github.com/metriport/metriport/pull/2960

### Description

- having ehr routes reflect new patient routes

### Testing

- Local
  - [x] EHR routes work
- Staging
  - [ ] EHR routes work
- Production
  - [ ] EHR routes work

### Release Plan

- [ ] Merge this

### CODE REVIEW

**Review by leite08 — COMMENTED** [review]
Can we add comment on the files below indicating the routes are used by other 'root" parent as well, please?
- `packages/api/src/routes/medical/index.ts`
- `packages/api/src/routes/medical/patient.ts`
- `packages/api/src/routes/medical/patient-root.ts`

**Comment by leite08** [comment]
Also, this should be an upstream dependency, right?

https://github.com/metriport/metriport/pull/2960

**Comment by thomasyopes** [comment]
> Also, this should be an upstream dependency, right?
> 
> #2960

Updated, but you also have this as an upstream, so https://github.com/metriport/metriport/pull/2960 needs to be updated (or we can reverse, they have to go together anyway).

File: packages/api/src/routes/ehr/util.ts [hunk 1]
```diff
@@ -6,25 +6,30 @@ import { EhrSources } from "../../external/ehr/shared";
...
 }
 
-export function parseIdFromPathParams(req: Request, regex: RegExp, regexIndex: number): string {
-  const matches = req.path.match(regex);
-  if (!matches) throw new BadRequestError("Request missing path param when required.");
-  const paramValue = matches[regexIndex];
-  if (!paramValue) throw new BadRequestError("Request missing path param when required.");
-  return paramValue;
+export function parseIdFromPathParams(req: Request, pathParamkey: string): string {
+  if (!req.params) throw new BadRequestError(`Request missing path param ${pathParamkey}`);
```
Comment (leite08) [inline packages/api/src/routes/ehr/util.ts:23]: This message should read something like "Request missing path parameters", right?

File: packages/api/src/routes/ehr/util.ts [hunk 2]
```diff
@@ -6,25 +6,30 @@ import { EhrSources } from "../../external/ehr/shared";
...
+  const pathParamValue = req.params[pathParamkey];
+  if (!pathParamValue) throw new BadRequestError(`Request missing path param ${pathParamkey}`);
+  const re = new RegExp(idRegex);
+  if (!re.test(pathParamValue)) {
+    throw new BadRequestError(
+      `Path param value for path param ${pathParamValue} is incorrectly formmated`
```
Comment (leite08) [inline packages/api/src/routes/ehr/util.ts:29]: nit: Could prob remove "path param" and just leave "Value for path param...", it seem to read better, no?

File: packages/api/src/routes/medical/index.ts [hunk 3]
```diff
@@ -16,8 +16,10 @@ routes.use("/facility", facilityRoot);
 routes.use("/facility/:id", facilityAuthorization("params"), facility);
 
 routes.use("/patient", patientRoot);
+// patient routes are also used in EHR Integrations
```
Comment (leite08) [inline packages/api/src/routes/medical/index.ts:19]: nit: "...also used in EHR Integrations routes"? Just to add a bit more context and help the reader to find the places being referenced

File: packages/api/src/routes/ehr/util.ts (hunk 1 above)
Comment (thomasyopes) [reply packages/api/src/routes/ehr/util.ts]: Done.

File: packages/api/src/routes/ehr/util.ts (hunk 2 above)
Comment (thomasyopes) [reply packages/api/src/routes/ehr/util.ts]: Done.

File: packages/api/src/routes/medical/index.ts (hunk 3 above)
Comment (thomasyopes) [reply packages/api/src/routes/medical/index.ts]: Done.

**Review by leite08 — APPROVED** [review]
🚀

**Pull request merged**